### PR TITLE
Get rid of interceptor injector and AOP from servlet configuration. (done)

### PR DESCRIPTION
**Pull request by akshaymayekar**
1. Changed default annotation handler to **RequestMappingHandlerMapping** from **DefaultAnnotationHandlerMapping**(It has been deprecated).
2. Moved **FlashInterceptor** to RequestMappingHandlerMapping interceptor and removed **InterceptorInjector** and **AOP**.

### CODE REVIEW

**Comment by AksMKR** [comment]
To be merged after url-rewrite PR is merged.

File: server/webapp/WEB-INF/spring-all-servlet.xml [hunk 1]
```diff
@@ -38,36 +37,20 @@
   <context:component-scan base-package="com.thoughtworks.go.server.web"/>
   <context:component-scan base-package="com.thoughtworks.go.server.newsecurity.controllers"/>
 
+  <mvc:annotation-driven/>
```
Comment (ketan) [inline server/webapp/WEB-INF/spring-all-servlet.xml]: I believe this is the default behavior with spring.

File: server/src/main/java/com/thoughtworks/go/server/web/InterceptorInjector.java [hunk 2]
```diff
@@ -1,50 +0,0 @@
...
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- *************************GO-LICENSE-END***********************************/
-
-package com.thoughtworks.go.server.web;
-
-import org.aspectj.lang.ProceedingJoinPoint;
```
Comment (ketan) [inline server/src/main/java/com/thoughtworks/go/server/web/InterceptorInjector.java]: Remove the aspectj dependency too?

File: server/webapp/WEB-INF/spring-all-servlet.xml (hunk 1 above)
Comment (AksMKR) [reply server/webapp/WEB-INF/spring-all-servlet.xml]: This is needed to instantiate RequestMappingHandlerMapping. For DefaultAnnotationHandlerMapping we don't need that tag.

File: server/src/main/java/com/thoughtworks/go/server/web/InterceptorInjector.java (hunk 2 above)
Comment (AksMKR) [reply server/src/main/java/com/thoughtworks/go/server/web/InterceptorInjector.java]: removed **aspectj** dependency. Keeping the **aspectjweaver** as its used at few places.

**Comment by maheshp** [comment]
Related to https://github.com/gocd/gocd/pull/4887, closing for now.

**Pull request closed without merging**